### PR TITLE
Fix(scripts/spells) charge Auto attack

### PR DESCRIPTION
**Pull request by pklloveyou**
## Changes Proposed:
-  
-  

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/11653

## Tests Performed:

- 
- 

## How to Test the Changes:

1.
2.
3.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/game/Movement/MotionMaster.cpp [hunk 1]
```diff
@@ -608,11 +608,11 @@ void MotionMaster::MoveCharge(float x, float y, float z, float speed, uint32 id,
     }
 }
 
-void MotionMaster::MoveCharge(PathGenerator const& path, float speed /*= SPEED_CHARGE*/)
+void MotionMaster::MoveCharge(PathGenerator const& path, float speed /*= SPEED_CHARGE*/, uint64 targetGUID /*= 0*/)
```
Comment (Kitzunu) [inline src/server/game/Movement/MotionMaster.cpp:611]: ObjectGuid

**Comment by pklloveyou** [comment]
@Kitzunu thanks

**Comment by pklloveyou** [comment]
image
It has been used in my project for a long time, which can ensure that it has no errors

File: src/server/game/Spells/Spell.cpp [hunk 2]
```diff
@@ -6005,7 +6005,7 @@ SpellCastResult Spell::CheckCast(bool strict)
                         if (!target->IsWithinLOSInMap(m_caster)) //Do full LoS/Path check. Don't exclude m2
                             return SPELL_FAILED_LINE_OF_SIGHT;
 
-                        float objSize = target->GetCombatReach();
+                        float objSize = m_caster->GetCombatReach() + target->GetCombatReach();
```
Comment (UltraNix) [inline src/server/game/Spells/Spell.cpp:6008]: Nope - it's has to be only target's combat reach.

File: src/server/game/Spells/SpellEffects.cpp [hunk 3]
```diff
@@ -4874,8 +4874,8 @@ void Spell::EffectCharge(SpellEffIndex /*effIndex*/)
         // Spell is not using explicit target - no generated path
         if (!m_preGeneratedPath)
         {
-            Position pos = unitTarget->GetFirstCollisionPosition(unitTarget->GetCombatReach(), unitTarget->GetRelativeAngle(m_caster));
-            m_caster->GetMotionMaster()->MoveCharge(pos.m_positionX, pos.m_positionY, pos.m_positionZ, speed);
+            Position pos = unitTarget->GetFirstCollisionPosition(m_caster->GetCombatReach() + unitTarget->GetCombatReach(), unitTarget->GetRelativeAngle(m_caster));
```
Comment (UltraNix) [inline src/server/game/Spells/SpellEffects.cpp:4877]: As above.

**Review by UltraNix — CHANGES_REQUESTED** [review]
What issue this PR resolves?

**Comment by pklloveyou** [comment]
@UltraNix --11670

**Comment by UltraNix** [comment]
So this PR is not a solution to this.

**Comment by pklloveyou** [comment]
They were deleted in 11534

**Comment by UltraNix** [comment]
59e45c251eb has broken that.

**Comment by pklloveyou** [comment]
I've deleted the scope of the battle

**Review by UltraNix — APPROVED** [review]
Now looks okay. 
Link it to issue.

**Comment by pklloveyou** [comment]
ok

**Comment by Kitzunu** [comment]
Did you test it @pklloveyou

**Comment by pklloveyou** [comment]
@Kitzunu yes

**Pull request merged**